SDWebImage/Core/SDAnimatedImageView.m

```diff
@@ -306,6 +306,19 @@ - (void)setHidden:(BOOL)hidden
 #pragma mark - UIImageView Method Overrides
 #pragma mark Image Data
 
+- (void)setAnimationRepeatCount:(NSInteger)animationRepeatCount
+{
+#if SD_UIKIT
+    [super setAnimationRepeatCount:animationRepeatCount];
+#else
+    _animationRepeatCount = animationRepeatCount;
+#endif
+    
+    if (self.shouldCustomLoopCount) {
```

Comment:
Good catch. Another issue during the refactory. Maybe v5.3.2 is soon.

Reply:
Do I need to add one more test case about this ? Avoid future regression issue.

```objective-c
SDAnimatedImageView *imageView;
imageView.image = animatedImage;
[imageView startAnimating];
imageView.shouldCustomLoopCount = YES;
imageView.animationRepeatCount = 100;
expect(imageView.player.totalLoopCount).equal(100);
```

Reply:
I think it's enough to set the animationrepeatcount property. But if we want to do better, when setting the new animationrepeatcount property, we should compare the size of the old and new values, and judge the current playing state. Theoretically, there should be six situations, but there may be only two that need to be dealt with, that is, the current playback is not over, when the new value is small (including unlimited playback), and when the current playback times are greater than the new value, the playback will be stopped immediately; the other is when the current playback is completed and the playback times are less than the new value, the playback will continue. This code is not RP, but I want to discuss whether to add this logic.

Reply:
> Do I need to add one more test case about this ? Avoid future regression issue.
> 
> ```objc
> SDAnimatedImageView *imageView;
> imageView.image = animatedImage;
> [imageView startAnimating];
> expect(imageView.player.totalLoopCount).equal(0);
> imageView.shouldCustomLoopCount = YES;
> imageView.animationRepeatCount = 100;
> expect(imageView.player.totalLoopCount).equal(100);
> ```

I think it's enough to set the animationrepeatcount property. But if we want to do better, when setting the new animationrepeatcount property, we should compare the size of the old and new values, and judge the current playing state. Theoretically, there should be six situations, but there may be only two that need to be dealt with, that is, the current playback is not over, when the new value is small (including unlimited playback), and when the current playback times are greater than the new value, the playback will be stopped immediately; the other is when the current playback is completed and the playback times are less than the new value, the playback will continue. This code is not RP, but I want to discuss whether to add this logic.

Reply:
Maybe not. This is a fix version, all previous behavior, just change the loop count, does not do anything related to the stop/start logic. Let's just keep the logic the same as before.

This test case can ensure `when setup the Animated ImageView.loopCount, we indeed update the player loop count`